lib/Conversion/TorchToStablehlo/Basic.cpp

```diff
@@ -432,21 +432,25 @@ class ConvertAtenMulDivOp : public OpConversionPattern<AtenOpT> {
     Value result =
         rewriter.create<ChloOpT>(loc, outType, lhs, rhs, bcastDimensions);
 
-    if (!isa<AtenDivTensorModeOp>(op)) {
+    if (!isa<AtenDivTensorModeOp, AtenDivScalarModeOp>(&op)) {
       rewriter.replaceOp(op, result);
       return success();
     }
 
-    AtenDivTensorModeOp divTensorModeOp =
-        llvm::dyn_cast<AtenDivTensorModeOp>(op.getOperation());
+    auto tensorOp = dyn_cast<AtenDivTensorModeOp>(op.getOperation());
+    auto opRoundingMode =
+        tensorOp
+            ? tensorOp.getRoundingMode()
+            : cast<AtenDivScalarModeOp>(op.getOperation()).getRoundingMode();
+
     std::string roundingMode;
-    if (!matchPattern(divTensorModeOp.getRoundingMode(),
-                      m_TorchConstantStr(roundingMode)))
+    if (!matchPattern(opRoundingMode, m_TorchConstantStr(roundingMode))) {
       return rewriter.notifyMatchFailure(
           op, "only support constant str rounding mode");
+    }
 
     // if trunc and int, do nothing
-    if (roundingMode == "trunc" && isa<mlir::FloatType>(outElemTy)) {
+    if (roundingMode == "trunc" && outElemTy.isa<mlir::FloatType>()) {
```

Comment:
@rsuderman you mentioned earlier that we want to write casts using the non-member function. Does that apply to `isa<>()`? Or does the opposite apply?

Reply:
I think we want to do the non-member version? IIRC it does a pointer case and some magic but its preferable due to better behavior for templated code. Similar comment at the top.